stripe/src/test/java/com/stripe/android/view/CardInputWidgetTest.kt

```diff
@@ -62,18 +64,49 @@ internal class CardInputWidgetTest : BaseViewTest<CardInputTestActivity>(
         activity = createStartedActivity()
 
         val dimensionOverrides = object : CardInputWidget.DimensionOverrideSettings {
+            override val frameWidth: Int
+                // That's a pretty small screen, but one that we theoretically support.
+                get() = 500
+
             override fun getPixelWidth(text: String, editText: EditText): Int {
                 // This makes it simple to know what to expect.
                 return text.length * 10
             }
-
-            override fun getFrameWidth(): Int {
-                // That's a pretty small screen, but one that we theoretically support.
-                return 500
-            }
         }
 
         cardInputWidget = activity.cardInputWidget
+
+        cardInputWidget.setCardNumberTextWatcher(object : TextWatcher {
+            override fun afterTextChanged(p0: Editable?) {
+            }
+
+            override fun beforeTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {
+            }
+
+            override fun onTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {
+            }
+        })
+        cardInputWidget.setExpiryDateTextWatcher(object : TextWatcher {
+            override fun afterTextChanged(p0: Editable?) {
+            }
+
+            override fun beforeTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {
+            }
+
+            override fun onTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {
+            }
+        })
+        cardInputWidget.setCvcNumberTextWatcher(object : TextWatcher {
+            override fun afterTextChanged(p0: Editable?) {
+            }
+
+            override fun beforeTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {
+            }
+
+            override fun onTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {
+            }
+        })
```

Comment:
nit: for cases like this I usually create an `EmptyTextWatcher` or `NoOpTextWatcher` implementation of the interface that's empty, so you can just use and avoid the duplication here

Reply:
good idea!